airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,548 @@
+    are used as credentials usually and you can specify different authentication parameters
+    via init params or via corresponding extras in the connection.
+
+    The extras in the connection are named the same as the parameters (`mount_point`,'kv_engine_version' ...).
+
+    Login/Password are used as credentials:
+
+        * approle: password -> secret_id
+        * ldap: login -> username,   password -> password
+        * userpass: login -> username, password -> password
+
+    :param vault_conn_id: The id of the connection to use
+    :type vault_conn_id: str
+    :param auth_type: Authentication Type for Vault. Default is ``token``. Available values are in
+        :py:const:`airflow.providers.hashicorp.hooks.vault.VALID_AUTH_TYPES`.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. Default is "secret". Note that
+         this ``mount_point`` is not used for authentication if authentication is done via a
+         different engine.
+    :type mount_point: str
+    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``). Defaults to
+          version defined in connection or ``2`` if not defined in connection.
+    :type kv_engine_version: int
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param role_id: Role ID for Authentication (for ``approle`` auth_type)
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+    :type gcp_key_path: str
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        vault_conn_id: str,
+        auth_type: Optional[str] = None,
+        mount_point: Optional[str] = None,
+        kv_engine_version: Optional[int] = None,
+        token: Optional[str] = None,
+        role_id: Optional[str] = None,
+        kubernetes_role: Optional[str] = None,
+        kubernetes_jwt_path: Optional[str] = None,
+        gcp_key_path: Optional[str] = None,
+        gcp_scopes: Optional[str] = None,
+    ):
+        super().__init__()
+        self.connection = self.get_connection(vault_conn_id)
+
+        if not auth_type:
+            auth_type = self.connection.extra_dejson.get('auth_type') or "token"
+
+        if not mount_point:
+            mount_point = self.connection.extra_dejson.get('mount_point')
+        if not mount_point:
+            mount_point = 'secret'
```

Comment:
Does it make sense to take the mountpoint from the "path" part of the Connection? (I think that is confusingly named as "connectin.schema")

That way the connection couldbe be defined as `vault://user:p@vault/my/secrets`. WDYT?

Reply:
Good Idea. I think it make sense especially that in the calls to Vault this is exactly how the calls are made.

Reply:
BTW. Will vault:// be usable in 1.10.* ? I will have to check it since I have to add it "Connection.py". I planned to use "http:"// originally (and I think it will work even in 1.10 anyway if I add it as http://) but I am wondering if I will be able to add vault:// as connection regardless. I will try it - it makes a different for backport documentation.

Reply:
I think the only thing that goes from connection prefix to Hook is a few of the DB hooks etc, as there are already (even in 1.10) many many mapping types that are not defined  in Connection.py list.